### PR TITLE
allow popen of action to take a shell argument

### PR DESCRIPTION
**Pull request by gaborbernat**
this allows plugin users to use our framework to handle shell calls :-)

### CODE REVIEW

**Comment by obestwalter** [comment]
Can you give a quick example in which situation this would be necessary? Generally I get a bit scared when seeing `shell=True` being used for anything, so there should be a good reason for adding this.

File: tox/session.py [hunk 1]
```diff
@@ -133,7 +134,16 @@ def _initlogpath(self, actionid):
...
-    def popen(self, args, cwd=None, env=None, redirect=True, returnout=False, ignore_ret=False):
+    def popen(
+        self,
+        args,
+        cwd=None,
+        env=None,
+        redirect=True,
+        returnout=False,
+        ignore_ret=False,
+        shell=False,
```
Comment (asottile) [inline tox/session.py]: I'm actually pretty `-1` on this.  shells are a great way to expose a CVE.  If someone wants a shell they can already `['bash', '-c', ...]`

**Comment by gaborbernat** [comment]
conda has some machinery (conda activate) that requires running in shell mode; here we don't run things ourselves in shell, but allow plugin users to do so without re-writing themselves this whole logic

**Comment by obestwalter** [comment]
And there no other way to accomplish this?

**Comment by asottile** [comment]
is `conda activate` more than just setting some environment variables `PATH`, `LD_..`, etc.?

**Comment by gaborbernat** [comment]
@asottile seems it's much more, it can run arbitrary user provided scripts

**Comment by gaborbernat** [comment]
any better ideas? :-)

**Comment by gaborbernat** [comment]
``['bash', '-c', ...]`` it is then :+1:

**Pull request closed without merging**